### PR TITLE
[Server] Filter permissions correctly when getting project summaries

### PR DESCRIPTION
**Pull request by adamdelman**
Filter permissions correctly when getting project summaries [ML-7393]  
Query OPA using a new resource type "project-summaries"

https://iguazio.atlassian.net/browse/ML-7665

[ML-7393]: https://iguazio.atlassian.net/browse/ML-7393?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: server/api/api/endpoints/projects.py [hunk 1]
```diff
@@ -376,12 +376,17 @@ async def list_project_summaries(
...
-        allowed_project_names = await server.api.utils.auth.verifier.AuthVerifier().filter_projects_by_permissions(
-            projects_output.projects,
-            auth_info,
+        allowed_project_names = await server.api.utils.auth.verifier.AuthVerifier().filter_project_resources_by_permissions(
+            resource_type=mlrun.common.schemas.AuthorizationResourceTypes.project_summaries,
+            resources=projects_output.projects,
+            project_and_resource_name_extractor=lambda project: (
+                project,
+                "",
+            ),
```
Comment (TomerShor) [inline server/api/api/endpoints/projects.py:389]: if you don't pass a `resource_name`, why not remove it from the template?

**Review by TomerShor — CHANGES_REQUESTED** [review]
Don't forget to run `make fmt lint` before you commit.

**Review by alonmr — APPROVED** [review]
Nice job! LGTM

File: server/api/api/endpoints/projects.py [hunk 2]
```diff
@@ -376,12 +376,18 @@ async def list_project_summaries(
         auth_info.projects_role,
         auth_info.session,
     )
-    allowed_project_names = projects_output.projects
```
Comment (alonmr) [inline server/api/api/endpoints/projects.py:379]: isn't this line still needed?

**Review by liranbg — APPROVED** [review]

**Pull request merged**